### ISSUE TITLE
Parcelable implementation for Deferred?

### ISSUE THREAD

**Issue by JvmName** [Backbase/DeferredResources#49]
**TLDR:** it'd be super useful for DeferredResources to extend `Parcelable`.

Here's my setup: 
- My ViewModels use a `val navigation: StateFlow<NavCommand>` to communicate with the UI layer, so that my Fragments don't have to concern themselves with the _what_ of navigation, just the _how_. 
- `NavCommand` is just a sealed class that wraps the different sort of operations one may take using the Nav component
```kotlin 
sealed class NavCommand {
    class To(val directions: NavDirections) : NavCommand()
    class Deeplink(...) : NavCommand()
    object Back : NavCommand()
    class BackTo(@IdRes val destinationId: Int, val inclusive: Boolean = false) : NavCommand()
}
```
- when the ViewModel decides navigation to another fragment needs to happen, typically it calls `navigation.value = To(SomethingFragmentDirections.actionFooToBar(a, b, c))`, where the inputs are the standard types that are acceptable in Bundles; one of those types is Parcelable. 
- at the UI layer, I say `val args by navArgs<NewFragmentArgs>()` (i.e. I use SafeArgs)

Because the Directions are instantiated at the VM-level, I can't `.resolve` the deferred resource at the time I (implicitly) build the Bundle.

Since I use safeargs, so as long as each concrete type knows how to serialize/deserialize from `Parcel`, I _think_ SafeArgs will code-gen the correct thing (i.e `DeferredText.Constant` won't come out the other side as `DeferredText.Resource`). 

Unfortunately, for people who _don't_ use SafeArgs, they'll have to manually remember to cast as `Constant` instead of `Resource`, otherwise they're gonna have a bad time.

**Comment by drewhamilton** [comment]
Thanks for filing an issue, seems like a legit use case. I think I'll set up some tests around your SafeArgs use case as well as a non-SafeArgs use case, and then see how far concrete-class-level Parcelable implementations get us.

If that falls short in some way then it's a less straightforward decision of whether/how to make the interfaces Parcelable.

Hoping to explore this within the next week.

**Comment by drewhamilton** [comment]
OK so as you suspected, making only the concrete classes `Parcelable` "works", but requires the receiving code to know the specific subtype being used if they're using a Bundle. This is true both for using a Bundle directly and for SafeArgs (where you'd have to put the concrete type in the navigation graph XML).

So we need a common interface for each type that implements `Parcelable`. Since all of the existing interfaces are already non-Parcelable and I don't want to force consumers to implement `Parcelable` if they don't need it, this means adding new types like `ParcelableDeferredText`. I'm a bit wary to basically double the number of interfaces this library ships, but I do think it's a valid use case and the correct approach, so I guess I'll do it anyway.

**Comment by drewhamilton** [comment]
Can you give `1.3.0-SNAPSHOT` a try and make sure it's working as expected in your project?

**Comment by JvmName** [comment]
sorry it took me so long -- just tested on `1.3.0-SNAPSHOT` using `DeferredFormattedString` and it worked! 🎉 

This is gonna make my life just a little bit easier, thank you!

**Comment by drewhamilton** [comment]
Thanks for testing it – [1.3.0](https://github.com/Backbase/DeferredResources/releases/tag/1.3.0) is released 🚀
